src/debugtopus.ts

```diff
@@ -25,6 +24,28 @@ export default defineConfig({
 });
 `;
 
+const getTempDirOnPackageRootLevel = (inputDir: string): string => {
+  let infiniteLoopPrevention = 5;
+  let appDir = inputDir;
+
+  while (infiniteLoopPrevention > 0) {
+    const nodeDir = path.join(appDir, "node_modules");
+    if (existsSync(nodeDir)) {
+      break;
+    }
+    appDir = path.join(appDir, "..");
+
+    infiniteLoopPrevention -= 1;
+  }
+
+  if (infiniteLoopPrevention === 0) {
+    throw new Error("can't find root level node modules :/");
+  }
+
+  const tempDir = path.join(appDir, "temp");
+  return tempDir;
+};
+
```

Comment:
would love to unit test this, just no idea how @Germandrummer92

Reply:
somehow mocking `existsSync` to return false X amount of time and then true 🤷

Reply:
yes thats probably the right idea, something like this should work:

```ts
jest.mock("fs");

let callCount = 0;
jest.mocked(existsSync).mockImplementation(() => {
    callCount++;
    if (callCount === 5) {
        return true;
    } 
    return false;
});
```